src/runtime/str.cpp

```diff
@@ -2805,7 +2805,7 @@ void setupStr() {
     str_cls->tp_flags |= Py_TPFLAGS_HAVE_NEWBUFFER;
 
     str_iterator_cls = BoxedClass::create(type_cls, object_cls, &BoxedStringIterator::gcHandler, 0, 0,
-                                          sizeof(BoxedStringIterator), false, "striterator");
+                                          sizeof(BoxedStringIterator), false, "iterator");
```

Comment:
I think the issue is not that we named this class wrong, but that it's not supposed to exist at all.  I had started to address this in #957, but it got held up because it caused a perf regression.  Long story short, I think we should leave the name the same here.